pkg/kp/rcstore/consul_store.go

```diff
@@ -301,7 +312,17 @@ func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{
 	errors := make(chan error)
 	input := make(chan *api.KVPair)
 
-	go consulutil.WatchSingle(kp.RCPath(rc.ID.String()), s.kv, input, quit, errors)
+	go func() {
+		rcp, err := RCPath(rc.ID)
+		if err != nil {
+			select {
+			case <-quit:
+			case errors <- err:
+			}
+			return
```

Comment:
@bwester what do you think about making this function return an `error` in addition to the other two return values. I know that's a really weird signature, but really the inability to compute this path is unrecoverable and the whole operation should be abandoned.

that would propagate to pkg/rc/farm.glo calling `WatchDesires` on a child. If an error is returned it would release its lock and allow another farm to pick up the replication controller.